x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/persistence/DataFrameTransformsConfigManager.java

```diff
@@ -335,6 +335,7 @@ public void getTransformStats(Collection<String> transformIds, ActionListener<Li
         SearchRequest searchRequest = client.prepareSearch(DataFrameInternalIndex.INDEX_NAME)
                 .addSort(DataFrameField.ID.getPreferredName(), SortOrder.ASC)
                 .setQuery(builder)
+                .setSize(transformIds.size() > 10_000 ? 10_000 : transformIds.size())
```

Comment:
[optional]
This can also be formulated as:
.setSize(Math.min(transformIds.size(), 10_000))